crates/bevy_pbr/src/render/pbr.wgsl

```diff
@@ -139,12 +162,37 @@ fn fragment(
         pbr_input.occlusion = occlusion;
 
         pbr_input.flags = mesh[in.instance_index].flags;
-
+#ifdef DEFERRED_PREPASS
+        pbr_functions::alpha_discard(pbr_bindings::material, output_color);
+        out.deferred = pbr_deferred_functions::deferred_gbuffer_from_pbr_input(pbr_input);
+        out.deferred_lighting_pass_id = pbr_bindings::material.deferred_lighting_depth_id;
```

Comment:
Good catch on the different names, I've updated it everywhere to use pass_id rather than depth_id. 

This is allows materials to select the deferred lighting pass they want to use. The deferred lighting pass runs using depth equals against this number so the fragment shader only runs for fragments with a matching ID.

Reply:
ok i see, `out.deferred_lighting_pass_id` is at location(3) which is hooked to the `deferred_lighting_pass_id` texture in the `DeferredPrepassNode`, and later copied to depth via the `CopyLightingId` node. got it thanks